### PR TITLE
Added TermsSetQuery

### PR DESCRIPTION
**Pull request by milroyfraser**
Implement #299

### CODE REVIEW

File: src/Query/TermLevel/TermsSetQuery.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    private $terms;
+
+    /**
+     * Constructor.
+     *
+     * @param string $field      Field name
+     * @param array  $terms      An array of terms
+     * @param array  $parameters Optional parameters
+     */
+    public function __construct($field, $terms, array $parameters = [])
```
Comment (einorler) [inline src/Query/TermLevel/TermsSetQuery.php]: This query does not have any point without `minimum_should_match_field`. Please make it a separate required parameter in the constructor

File: src/Query/TermLevel/TermsSetQuery.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function toArray()
+    {
+        $query = [
+            'terms' => $this->terms,
+        ];
+
+        $output = $this->processArray($query);
```
Comment (einorler) [inline src/Query/TermLevel/TermsSetQuery.php]: there's no point in having this variable, could you just pass the `$this->processArray($query);` directly in the needed place of the return array

File: src/Query/TermLevel/TermsSetQuery.php (hunk 1 above)
Comment (milroyfraser) [reply src/Query/TermLevel/TermsSetQuery.php]: @einorler it can `minimum_should_match_field ` or `minimum_should_match_script` 🤔

Comment (milroyfraser) [reply src/Query/TermLevel/TermsSetQuery.php]: @einorler what is your thoughts on making `$parameters` a required parameter in the constructor and checking for `minimum_should_match_field` or `minimum_should_match_script` existence before calling setParameters.

**Comment by milroyfraser** [comment]
@saimaz any remarks on this ☝️ 
Tnx

**Comment by saimaz** [comment]
Seems all good, let's merge it

**Comment by saimaz** [comment]
cherry-picked in the https://github.com/ongr-io/ElasticsearchDSL/commit/59f9e4452f9762041e955d21725db7cce6510886

**Pull request closed without merging**